lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -98,25 +98,53 @@ const SessionAuth: React.FC<PropsWithChildren<SessionAuthProps>> = ({ children,
                 userId: "",
             };
         }
-        const invalidClaims = await session.current.validateClaims({
-            overrideGlobalClaimValidators: props.overrideGlobalClaimValidators,
-            userContext,
-        });
+
+        let invalidClaims;
+        try {
+            invalidClaims = await session.current.validateClaims({
+                overrideGlobalClaimValidators: props.overrideGlobalClaimValidators,
+                userContext,
+            });
+        } catch (ex) {
+            // These errors should only come from getAccessTokenPayloadSecurely inside validateClaims if refreshing a claim cleared the session
+            // Which means that the session was actually invalid meaning returning false is right.
+            // This might also happen if the user provides an override or a custom claim validator that throws (or if we have a bug)
+            // In one case it's the right way to handle this, and the other is a user error (and it should be rare)
+            return {
+                loading: false,
+                doesSessionExist: false,
+                accessTokenPayload: {},
+                invalidClaims: [],
+                userId: "",
+            };
```

Comment:
if a session exists, then we should throw ex, else we should return doesSessionExist false